compiler/frontend/beam/src/main/java/org/apache/nemo/compiler/frontend/beam/transform/GroupByKeyAndWindowDoFnTransform.java

```diff
@@ -179,7 +199,7 @@ private void triggerTimers(final K key,
     final List<TimerInternals.TimerData> timerDataList = getEligibleTimers(timerInternals);
 
     if (timerDataList.isEmpty()) {
-      return;
+      return Long.MAX_VALUE;
     }
```

Comment:
else {

}

To make it more explicit that when there are eligible timers we do `processElement` and `advanceOutputWatermark`.